microprofile/metrics-smallrye/src/main/java/org/wildfly/extension/microprofile/metrics/deployment/DeploymentMetricService.java

```diff
@@ -65,9 +66,9 @@ public void stop(StopContext stopContext) {
 
     private static PathAddress createDeploymentAddressPrefix(DeploymentUnit deploymentUnit) {
         if (deploymentUnit.getParent() == null) {
-            return PathAddress.pathAddress(DEPLOYMENT, deploymentUnit.getName());
+            return PathAddress.pathAddress(DEPLOYMENT, deploymentUnit.getAttachment(Attachments.MANAGEMENT_NAME));
         } else {
-            return createDeploymentAddressPrefix(deploymentUnit.getParent()).append(SUBDEPLOYMENT, deploymentUnit.getName());
+            return createDeploymentAddressPrefix(deploymentUnit.getParent()).append(SUBDEPLOYMENT, deploymentUnit.getAttachment(Attachments.MANAGEMENT_NAME));
```

Comment:
I'm pretty sure removing the change to this line will fix the problem.

I don't think a subdeployment has this attachment, and in any case a subdeployment never has a management name that is different from deploymentUnit.getName(). A user can't assign a management name to a subdeployment as it's an internal part of the top level deployment, not something independently added to the server config.